genisys/modules/preseed.py

```diff
@@ -3,52 +3,61 @@
 import jinja2
 from genisys.modules.base import Module
 from genisys.config_parser import YAMLParser
+from genisys.server import tls
 import sys
-FILENAME = "preseed.cfg"
 
+FILENAME = "preseed.cfg"
 class Preseed(Module):
     """Generates a Preseed file to be served over the network to a booting Debian system"""
     def __init__(self: Self, config: YAMLParser):
         self.config = {}
         self.config["network"] = config.get_section("Network")
         self.config["users"] = config.get_section("Users")
         self.config["apps"] = config.get_section("Applications")
-    # end __init__
 
     def install_location(self: Self) -> Path:
         """Places the Preseed file in the TFTP root"""
         ftp_dir = Path(self.config["network"]["tftp_directory"])
         return Path(ftp_dir, FILENAME)
-    # end install_location
 
     def generate(self: Self) -> str:
         """Renders the Jinja template with the prespecified configurations"""
         loader = jinja2.PackageLoader('genisys')
         jinja_env = jinja2.Environment(loader=loader)
         template = jinja_env.get_template("preseed.cfg.jinja2")
 
-        # convert booleans to lowercase to match preseed format
+        # Convert booleans to lowercase to match preseed format
         for key, value in self.config["users"].items():
             if isinstance(value, bool):
                 self.config["users"][key] = str(value).lower()
 
+                # Read SSH key files and store their contents as strings
+                ssh_keys_contents = []
+                for ssh_key_path_str in self.config["users"].get("ssh-keys", []):
+                    ssh_key_path = Path(ssh_key_path_str)  # Directly use the provided absolute path
+                    try:
+                        with open(ssh_key_path, 'r') as f:
+                            ssh_keys_contents.append(f.read())
+                    except FileNotFoundError:
+                        print(f"Warning: SSH key file {ssh_key_path} not found.")
 
-        return template.render(settings=self.config["users"], ftp=self.config["network"].get("ftp", {}))
-    # end generate
+                # Determine SSL certificate path
+                ssl_cert_path = Path(tls.get_keychain(self.config["network"]["server"]["ssl"])["certfile"])
```

Comment:
If you move this into the `if 'ssl' in self.config["network"].get("server", {}):` block on line 49 you ensure that the ssl key exists before accessing it which should fix the tests.